### ISSUE TITLE
VonMises standard deviation appears to be wrong

### ISSUE THREAD

**Issue by itsdfish** [JuliaStats/Distributions.jl#1705]
Hi,

I noticed that the VonMises standard deviation is wrong. Consider the following:

```
using Distributions
using Random 

Random.seed!(65)

dist = VonMises(0.0, 2.0)
x = rand(dist, 100_000)
@show std(x)
@show std(dist)
```

**Output**

```
std(x) = 0.8741380214035047
std(dist) = 0.5497502542391335
```

**Version**

```
Distributions v0.25.86
Julia 1.9.0 rc2
```

**Comment by andreasnoack** [comment]
> Could this be the reason for the discrepancy?

Yes. That is the difference. See https://en.wikipedia.org/wiki/Directional_statistics.

```julia
julia> sqrt(1 - abs(mean(exp.(complex.(0.0, x)))))
0.5492460106924333
```

Indeed, this could be better documented, so let's keep this issue open to track that. I've adjusted the title accordingly.

**Comment by ParadaCarleton** [comment]
> * Keep var as-is, fix std, update docs to refer to circular statistics

I believe this is the intention; I'd be happy to review a PR on this if mentioned.

**Comment by itsdfish** [comment]
I can submit a PR to make a note about circular statistics. I'm not sure where to update the docs, but I identified at least three places where the docs could be updated. @ParadaCarleton, do you prefer any or all of these?  Please advise. 


https://github.com/JuliaStats/Distributions.jl/blob/2cd6c678a1f237462d71496c39178dab1e265c61/src/univariates.jl#L175-L179

https://github.com/JuliaStats/Distributions.jl/blob/2cd6c678a1f237462d71496c39178dab1e265c61/src/univariate/continuous/vonmises.jl#L58

https://github.com/JuliaStats/Distributions.jl/blob/2cd6c678a1f237462d71496c39178dab1e265c61/src/univariates.jl#L175-L179

**Comment by itsdfish** [comment]
Sorry. I misunderstood that `std` has an error. I assumed `std(d) = sqrt(var(d))`.

**Comment by mateuszbaran** [comment]
> Technically, the latter seems more consistent with other distributions... It can potentially be confusing if `std(d) != std(rand(d, N))`.
> But are there actual scenarios when non-circular std/var of VonMises are needed?

I find using circular variance for for circular distributions confusing. Non-circular std/var is reasonable to use when the distribution is (approximately) supported on an arc.

Von-Mises Fisher distribution can be defined on any-dimensional sphere and has a valid, general notion of variance: http://www.math.ualberta.ca/~thillen/paper/vonmises.pdf . Were it implemented in Distributions.jl, should it use circular or generic definition when the sphere happens to be a circle?
